client/src/utils/configUtils.ts

```diff
@@ -1,12 +1,31 @@
 import { InspectorConfig } from "@/lib/configurationTypes";
 import { DEFAULT_MCP_PROXY_LISTEN_PORT } from "@/lib/constants";
 
+declare global {
+  interface Window {
+    __RUNTIME_CONFIG__?: {
+      MCP_PROXY_FULL_ADDRESS?: string;
+      MCP_PROXY_PORT?: string;
+    };
+  }
+}
+
 export const getMCPProxyAddress = (config: InspectorConfig): string => {
+  // First check for runtime injected MCP_PROXY_FULL_ADDRESS
+  if (window.__RUNTIME_CONFIG__?.MCP_PROXY_FULL_ADDRESS) {
+    return window.__RUNTIME_CONFIG__.MCP_PROXY_FULL_ADDRESS;
+  }
+
+  // Then check for config from local storage
   const proxyFullAddress = config.MCP_PROXY_FULL_ADDRESS.value as string;
   if (proxyFullAddress) {
     return proxyFullAddress;
   }
-  return `http://${window.location.hostname}:${DEFAULT_MCP_PROXY_LISTEN_PORT}`;
+
+  // Finally use the runtime port if available, otherwise default port
+  const proxyPort =
+    window.__RUNTIME_CONFIG__?.MCP_PROXY_PORT || DEFAULT_MCP_PROXY_LISTEN_PORT;
+  return `http://${window.location.hostname}:${proxyPort}`;
```

Comment:
There will be a conflict here. we have a merged change that gets the http: part from the `window.location.protocol`

Reply:
```suggestion
  return `${window.location.protocol]://${window.location.hostname}:${proxyPort}`;
```

I'll add that into the PR as well then merge `main` up and that should avoid the conflict.